File: java/src/Leetcode345.java
public class Leetcode345 {
    public String reverseVowels(String s) {        
        Stack<Character> stack = new Stack<>();
        char[] charArray = s.toCharArray();
        for(int i = 0; i < charArray.length; i++){
            if(isVowel(charArray[i]))
                stack.push(charArray[i]);
        }

        for(int i = 0; i < charArray.length; i++){
            if(isVowel(charArray[i]))
                charArray[i] = stack.pop();
        }
        return new String(charArray);
    }

    private boolean isVowel(char c){
        char[] vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'O', 'I', 'U'};
        for(char cha : vowels){
            if(cha == c)
                return true;
        }
        return false;
    }
}

File: python/Leetcode345.py
class Leetcode345:
    def reverseVowels(self, s: str) -> str:
        st = []
        vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        for c in s:
            if c in vowels:
                st.append(c)
        s_list = list(s)
        for i in range(len(s_list)):
            if s_list[i] in vowels:
                s_list[i] = st.pop()
        return ''.join(s_list)